R/smoking-status.R

```diff
@@ -0,0 +1,100 @@
+#' @title Simple smoking status
+#'
+#' @description This function creates a derived smoking variable (smoke_simple)  
+#'  with four categories: 
+#'  
+#' \itemize{
+#'   \item non-smoker (never smoked)
+#'   \item current smoker (daily and occasional?)
+#'   \item former daily smoker quit ≤5 years or former occasional smoker 
+#'   \item former daily smoker quit >5 years
+#'  }
+#'
+#' @param SMKDSTY derived variable that classifies an individual's smoking
+#'  status.
+#'
+#' @param SMK_09A_B number of years since quitting smoking. Variable asked to
+#'  former daily smokers who quit <3 years ago.
+#'
+#' @param SMKG09C number of years since quitting smoking. Variable asked to
+#'  former daily smokers who quit >=3 years ago.
+#'
+#' @example
+#' # Using the 'derive_smoke_simple' function to create the derived smoking   
+#' # variable across CCHS cycles.
+#' # derive_smoke_simple() is specified in the variable_details.csv
+#'
+#' # To create a harmonized smoke_simple variable across CCHS cycles, use 
+#' # rec_with_table() for each CCHS cycle and specify derive_smoke_simple and 
+#' # the required base variables.
+#' # Using bind_rows(), you can combine smoke_simple across cycles
+#'
+#' library(cchsflow)
+#'
+#' smoke_simple2009_2010 <- rec_with_table(
+#'   cchs2009_2010_p, c(
+#'     "SMKDSTY", "SMK_09A_B", "SMKG09C", "derive_smoke_simple"
+#'   )
+#' )
+#'
+#' head(smoke_simple2009_2010)
+#'
+#' smoke_simple2011_2012 <- rec_with_table(
+#'   cchs2011_2012_p,c(
+#'     "SMKDSTY", "SMK_09A_B", "SMKG09C", "derive_smoke_simple"
+#'   )
+#' )
+#'
+#' tail(smoke_simple2011_2012)
+#'
+#' combined_smoke_simple <- suppressWarnings(bind_rows(smoke_simple2009_2010,
+#'  smoke_simple2011_2012))
+#'
+#' head(combined_smoke_simple)
+#' tail(combined_smoke_simple)
+#' @export
+derive_smoke_simple <-
+  function(SMKDSTY, SMK_09A_B, SMKG09C) {
+    
+    # Nested function: current smoker status
+    derive_current_smoker <- function(SMKDSTY) {
+      smoker <-
+        if_else2(SMKDSTY %in% c(1, 2, 3), 1,
+                if_else2(SMKDSTY %in% c(4, 5, 6), 0, NA))
+      return(smoker)
+    }
+    smoker <- derive_current_smoker(SMKDSTY)
+    
+    # Nested function: ever smoker status
+    derive_ever_smoker <- function(SMKDSTY) {
+      eversmoker <-
+        if_else2(SMKDSTY %in% c(1, 2, 3, 4, 5), 1,
+                if_else2(SMKDSTY == 6, 0, NA))
+      return(eversmoker)
+    }
+    eversmoker <- derive_ever_smoker(SMKDSTY)
+    
+    # Nested function: time since quit former daily smokers
+    tsq_ds_fun <- function(SMK_09A_B, SMKG09C) {
+      SMKG09C <-
```

Comment:
Ahhh...good point. I used the same code that @wyusuf068 used for the pack years function. Any change that is made should be done to both functions. This also raises a question about accessing nested functions within cchsflow. Given Warsame already created tsq_ds_fun within pack_years_der, could I have just called this function?

Reply:
@yulric and @wyusuf068 I've revised the nested function so that we are not mutating SMKG09C for tsq_ds_fun. This method should be made consistent in pack_years_fun.
Also, do we need to be re-specifying this function? (see question above). I don't think so...for the very reason we are faced with now (i.e., harmonizing the change in smoke_simple_fun and pack_years_fun)

Reply:
`tsq_ds_fun()` looks good to me. One thing I was unsure was what is the return object? I'm assuming `tsq_ds`. A return line for `tsq_ds` would be helpful to prevent unwanted errors.